enhancements/update/versioning-cincinnati-api.md

```diff
@@ -0,0 +1,116 @@
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+
+## Summary
+As of today, we declare the schema version in Cincinnati Graph URI. We currently, do not embed the version information
+in json schema. The enhancement will embed the version information in the json schema. This has the benefit of being 
+easy to implement and to pass around if we ship Cincinnati JSON over non-HTTP channels. Enhancement will also implement
+media type versioning while requesting the graph e.g. `ContentType: application/vnd.redhat.cincinnati.v1+json`. 
+The move to supplying requested version in media type field will require us to move towards a generic un-versioned Graph
+URI, with the move, we'll still support old URI for foreseeable future. 
+
+## Motivation
+
+### Goals
+1. Embed version information in json schema.
+2. Implement media type versioning for clients requesting a specific schema version of cincinnati
+3. Move to a generic un-versioned URI for cincinnati
+4. Cincinnati continues to support older clients after adding support for new clients for foreseeable future.
+5. Add metrics to identify hits on URIs and versions.
+
+### Non-Goals
+
+## Proposal
+
+### Implementation
+
+#### Version information in json schema 
+Embed the version information in json schema. The updated json schema should look like:
+```json
+{
+    "version": 1,
+    "nodes": [],
+    "edges": []
+}
+```
+
+#### Media Type Versioning 
+Currently, the versioning is done by using the versioned URI and does not support media type versioning. The user should 
+be able to request a specific version of cincinnati graph by supplying the version in media type field of the request. 
+for e.g. `ContentType: application/vnd.redhat.cincinnati.v1+json`. Here are some of the 
+[examples of media type versioning](https://github.com/opencontainers/image-spec/blob/v1.0.1/media-types.md).       
+In the event where the media type is not provided, or `application/json` is provided, cincinnati should return a default 
+response that is compatible with earlier versions (v1) so as not to break old clusters from upgrading. 
+
+
+#### URI Change
+The current URI follows `{URL}/v1/graph`. The new un-versioned URI will follow `{URL}/graph`. The existing URI can be 
+redirected to the new URI. This can be done through cincinnati. In the case where an old URI is being called, cincinnati 
+will by default return the graph that is compatible with `v1`.      
+
+
+#### Version metrics
+Cincinnati should also implement a way to understand how many clients are consuming the various versions. 
+This can be done by implementing metrics for old URI vs new URI. Similarly, metrics should be implemented to understand
+the consumption of various media types (versions). This will help us to make decisions on deprecating the old versions 
+and URI. 
```

Comment:
do we want to spell this out in more detail?  I'm not sure if the host cluster's Router gives us any metrics we can use out of the box.  But if we need out own, new metrics, probably a single `cincinnati_pe_http_response_total` counter with labels for `uri` and `version`?  Or something...